pulpcore/pulpcore/app/serializers/base.py

```diff
@@ -232,8 +232,7 @@ def to_representation(self, instance):
         ret = OrderedDict()
 
         instance = instance.cast()
-        viewset = viewset_for_model(instance)
-        fields = viewset.serializer_class(context=self._context)._readable_fields
```

Comment:
I need some special attention paid to this line.  I think it should be fine but just want to be sure.

The reason this line is problematic is because we want to be able to redefine which serializer class is being used at runtime based on the action being exercised (e.g. use a different, stripped-down serializer for listing objects).  DRF is fine with that, which is why internally they *never* use ```.serializer_class``` and instead use ```get_serializer_class()``` everywhere - explicitly so that people can redefine that behavior.

Reply:
And FWIW, I don't understand what this method is doing.  I commented the whole thing out and as far as I could tell everything worked fine and exactly the same, and I tested a number of things.

Reply:
viewset is just a class, not an instance